### ISSUE TITLE
Feature request: Port forward for all ready resources

### ISSUE THREAD

**Issue by Richard87** [GoogleContainerTools/skaffold#5318]
I don't want skaffold to wait for all resources to be ready before starting to port-forward

This would make it easier to troubleshoot broken builds in our case :) 

### Expected behavior
Open port-forwarding when resource is ready

### Actual behavior
Waits for all resources to be ready before port-forwarding

### Information

- Skaffold version: 1.18
- Operating system: Fedora 33

**Comment by MarlonGamez** [comment]
Hi @Richard87, thanks for opening up this issue! This definitely seems like a reasonable request, and I'll discuss it with our team. It seems like it would require a bit of restructuring about how things are currently done within skaffold, but I'll update here once I have more information :)

**Comment by MarlonGamez** [comment]
Hey again, could you expand a bit more on your use case and how this functionality would help you? Does disabling the status check phase in skaffold by using `--status-check=false` help things?

**Comment by Richard87** [comment]
Hi! I was not aware of `--status-check=false`, I'll enable it and see what happens next time something breaks :) 

For example, if there is a problem with the database schema, our web-app doesn't go "ready", therefore port-forwarding doesn't start, and it gets much harder to connect to the database to fix the issue, and sometimes, Skaffold gives up and deletes all resources before I finish fixing the issue.

Or another example, if someone (probably me) have to work on different microservices at the same time, and havent checked out a different branch for each, *and* breaks one of the builds, I can't fix one at a time to get it up, everything have to be fixed before anything is available...

**Comment by Richard87** [comment]
Hmm, `--status-check=false` did bring it up without deleting the deployments, *but* no port-forwarding worked.

I got a whole bunch of `port forwarding service-eportal-db-richard-3306 got terminated: output: error: unable to forward port because pod is not running. Current status=Pending`

**Comment by tejal29** [comment]
For user usecase i think it makes sense to port-forward as resources are ready.

Thanks for the request @Richard87 . We will try to bundle it with #4832

**Comment by Richard87** [comment]
> For user usecase i think it makes sense to port-forward as resources are ready.
> 
> Thanks for the request @Richard87 . We will try to bundle it with #4832

So, port-forwarding starts as soon as the requested resource is ready, or will it wait until all resources are ready?

**Comment by Richard87** [comment]
version 1.22 /  #4832 did not solve my use case...

Skaffold still waits for all resources to be ready before port-forwarding any resources, even ready resources.
(I have one resource that takes some time to come up, and it would be nice to start working in dev since it's not required)

**Comment by gsquared94** [comment]
@Richard87 you're right, #4832 only modified the pod selection logic and doesn't really help with your issue. The code currently has the `status-check` and `port-forward` as separate stages so it's not a straight-forward refactor to make them triggerable with individual resource updates. I know that @nkubala is working on making the `port-forwarder` an element of the `Deployer`. Since `status-check` is already inside the `Deployer`, maybe this is something that could be attempted then?

**Comment by Richard87** [comment]
Thanks for keeping this on the radar, it would make my bad (coding) days way better! :)
